src/seedu/addressbook/Main.java

```diff
@@ -105,12 +116,14 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) throws StorageOperationException, RuntimeException  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            throw e;
```

Comment:
Is there a need to catch the storage exception here and throw it again when the executeCommand method is inside the try block where it is called?